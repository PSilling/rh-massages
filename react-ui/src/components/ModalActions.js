// react imports
import React, { Component } from 'react';

// util imports
import _t from '../utils/Translations.js';

/**
 * Custom component rendering action buttons for modal footers.
 */
class ModalActions extends Component {

  render() {
    return(
      <div className="pull-right">
        <button type="button" className="btn btn-primary" onClick={this.props.onProceed}
          style={{ 'marginRight': '5px' }} autoFocus={this.props.autoFocus}>
          {this.props.primaryLabel}
        </button>
        <button type="button" className="btn btn-default"
          onClick={this.props.onClose}>
          { _t.translate('Dismiss') }
        </button>
      </div>
    );
  }
}

export default ModalActions
