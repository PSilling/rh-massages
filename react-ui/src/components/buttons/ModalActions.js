// react imports
import React from "react";
import PropTypes from "prop-types";

// util imports
import _t from "../../util/Translations";

/**
 * Footer buttons for Modal dialogs. Children components can be given to add more
 * case specific elements.
 */
const ModalActions = function ModalActions(props) {
  return (
    <div className="pull-right">
      {props.children}
      {props.primaryLabel === "none" ? (
        ""
      ) : (
        <button
          type="button"
          className="btn btn-primary"
          onClick={props.onProceed}
          title={props.title}
          style={{ marginRight: "5px" }}
          autoFocus={props.autoFocus}
          disabled={props.disabled}
        >
          {props.primaryLabel}
        </button>
      )}
      <button type="button" className="btn btn-default" onClick={props.onClose}>
        {_t.translate("Dismiss")}
      </button>
    </div>
  );
};

ModalActions.propTypes = {
  /** callback function triggered on close button click */
  onClose: PropTypes.func.isRequired,
  /** callback function triggered on primary button click */
  onProceed: PropTypes.func.isRequired,
  /** whether the primary button should be automatically focused */
  autoFocus: PropTypes.bool,
  /** extra elements to be provided before the two buttons */
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node]),
  /** whether the primary button should be disabled */
  disabled: PropTypes.bool,
  /** primary button label to be displayed */
  primaryLabel: PropTypes.string,
  /** primary button title */
  title: PropTypes.string
};

ModalActions.defaultProps = {
  autoFocus: false,
  children: null,
  disabled: false,
  primaryLabel: _t.translate("Proceed"),
  title: ""
};

export default ModalActions;
