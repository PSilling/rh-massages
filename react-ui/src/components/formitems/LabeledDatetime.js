// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Col, FormGroup, Label, Tooltip } from "reactstrap";
import Datetime from "react-datetime";
import moment from "moment";

// util imports
import Util from "../../util/Util";

/**
 * A combined formgroup component with a label (can have a tooltip) and a Datetime input.
 * A column is included to set input size.
 */
class LabeledDatetime extends Component {
  state = { tooltipActive: false };

  tooltipTarget = Util.getTooltipTargets(1)[0];

  toggleTooltip = () => {
    this.setState(prevState => ({ tooltipActive: !prevState.tooltipActive }));
  };

  handleKeyPress = event => {
    if (event.key === "Enter") {
      this.props.onEnterPress();
    }
  };

  render() {
    const { disabled, onChange, label, value, labelWidth, onEnterPress, size, tooltip, ...rest } = this.props;
    return (
      <Col md={this.props.size}>
        <FormGroup>
          <Label id={this.tooltipTarget} for={`${this.tooltipTarget}_input`} style={{ width: this.props.labelWidth }}>
            {this.props.label}
          </Label>
          <Datetime
            {...rest}
            value={this.props.value}
            onChange={this.props.onChange}
            inputProps={{
              id: `${this.tooltipTarget}_input`,
              disabled,
              placeholder: this.props.label,
              onKeyPress: this.handleKeyPress
            }}
          />
          {this.props.tooltip !== "" && (
            <Tooltip isOpen={this.state.tooltipActive} target={this.tooltipTarget} toggle={this.toggleTooltip}>
              {this.props.tooltip}
            </Tooltip>
          )}
        </FormGroup>
      </Col>
    );
  }
}

LabeledDatetime.propTypes = {
  /** function called on input value change */
  onChange: PropTypes.func.isRequired,
  /** input label text */
  label: PropTypes.node.isRequired,
  /** current value of the input field */
  value: PropTypes.oneOfType([PropTypes.instanceOf(moment), PropTypes.instanceOf(Date)]).isRequired,
  /** whether the Datetime should be disabled */
  disabled: PropTypes.bool,
  /** width of the label container */
  labelWidth: PropTypes.string,
  /** input datalist options */
  onEnterPress: PropTypes.func,
  /** input column length */
  size: PropTypes.string,
  /** input label tooltip */
  tooltip: PropTypes.string
};

LabeledDatetime.defaultProps = {
  onEnterPress() {},
  disabled: false,
  labelWidth: "auto",
  size: "12",
  tooltip: ""
};

export default LabeledDatetime;
