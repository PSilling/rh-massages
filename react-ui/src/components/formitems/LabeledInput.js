// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Col, FormGroup, Label, Input, Tooltip } from "reactstrap";

// util imports
import Util from "../../util/Util";

/**
 * A combined formgroup component with a label (can have a tooltip) and a input.
 * A column is included to set input size.
 */
class LabeledInput extends Component {
  state = { tooltipActive: false };

  tooltipTarget = Util.getTooltipTargets(1)[0];

  toggleTooltip = () => {
    this.setState(prevState => ({ tooltipActive: !prevState.tooltipActive }));
  };

  handleKeyPress = event => {
    if (event.key === "Enter") {
      this.props.onEnterPress();
    }
  };

  render() {
    const { onChange, label, value, onEnterPress, options, size, tooltip, ...rest } = this.props;
    return (
      <Col md={this.props.size}>
        <FormGroup>
          <Label id={this.tooltipTarget} for={`${this.tooltipTarget}_input`}>
            {this.props.label}
          </Label>
          <Input
            {...rest}
            id={`${this.tooltipTarget}_input`}
            value={this.props.value}
            onChange={this.props.onChange}
            onKeyPress={this.handleKeyPress}
            placeholder={this.props.label}
            list={`${this.tooltipTargets}_list`}
          />
          {this.props.options.length > 0 && (
            <datalist id={`${this.tooltipTargets}_list`}>
              {this.props.options.map(item => (
                <option key={item} value={item} />
              ))}
            </datalist>
          )}
          {this.props.tooltip !== "" && (
            <Tooltip isOpen={this.state.tooltipActive} target={this.tooltipTarget} toggle={this.toggleTooltip}>
              {this.props.tooltip}
            </Tooltip>
          )}
        </FormGroup>
      </Col>
    );
  }
}

LabeledInput.propTypes = {
  /** function called on input value change */
  onChange: PropTypes.func.isRequired,
  /** input label text */
  label: PropTypes.string.isRequired,
  /** current value of the input field */
  value: PropTypes.string.isRequired,
  /** input datalist options */
  onEnterPress: PropTypes.func,
  /** input datalist options */
  options: PropTypes.arrayOf(PropTypes.string),
  /** input column length */
  size: PropTypes.string,
  /** input label tooltip */
  tooltip: PropTypes.string
};

LabeledInput.defaultProps = {
  onEnterPress() {},
  options: [],
  size: "12",
  tooltip: ""
};

export default LabeledInput;
