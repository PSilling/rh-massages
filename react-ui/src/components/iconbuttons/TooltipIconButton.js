// react imports
import React, { Component } from "react";
import PropTypes from "prop-types";

// module imports
import { Button, Tooltip } from "reactstrap";

// util imports
import Util from "../../util/Util";

/**
 * Labeless icon only button with an attached tooltip.
 */
class TooltipIconButton extends Component {
  state = { tooltipActive: false };

  tooltipTarget = Util.getTooltipTargets(1)[0];

  toggleTooltip = () => {
    this.setState(prevState => ({ tooltipActive: !prevState.tooltipActive }));
  };

  render() {
    const { disabled, icon, ...rest } = this.props;
    return (
      <span className="no-print">
        <Button {...rest} id={this.tooltipTarget} outline disabled={this.props.disabled}>
          <span className={`fas fa-${this.props.icon}`} />
        </Button>
        {this.props.tooltip !== "" && !this.props.disabled && (
          <Tooltip
            isOpen={this.state.tooltipActive}
            target={this.tooltipTarget}
            toggle={this.toggleTooltip}
            trigger="hover"
          >
            {this.props.tooltip}
          </Tooltip>
        )}
      </span>
    );
  }
}

TooltipIconButton.propTypes = {
  /** icon to be shown instead of a normal label */
  icon: PropTypes.string.isRequired,
  /** function to be called on button click */
  onClick: PropTypes.func.isRequired,
  /** whether the button should be disabled */
  disabled: PropTypes.bool,
  /** button size class */
  size: PropTypes.string,
  /** button component style */
  style: PropTypes.shape({
    border: PropTypes.string
  }),
  /** tooltip to be shown on button hover */
  tooltip: PropTypes.string
};

TooltipIconButton.defaultProps = {
  disabled: false,
  size: "sm",
  style: { border: "0px solid transparent" },
  tooltip: ""
};

export default TooltipIconButton;
