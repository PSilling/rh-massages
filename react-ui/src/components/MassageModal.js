// react imports
import React, { Component } from 'react';


// component imports
import AddButton from './AddButton';
import ModalActions from './ModalActions';

// module imports
import {ModalContainer, ModalDialog} from 'react-modal-dialog';
import moment from 'moment';
import DatePicker from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';

// util imports
import Auth from '../utils/Auth.js';
import _t from '../utils/Translations.js';
import Util from '../utils/Util.js';

class MassageModal extends Component {

  state = {date: moment(), masseuse: ""}

  /**
   * Sets default input values on props change.
   */
  componentWillReceiveProps(nextProps) {
    if(this.props === nextProps) return;

    this.setState({
      date: (nextProps.massage === -1) ? moment() : moment(nextProps.massage.date),
      masseuse: (nextProps.massage === -1) ? "" : nextProps.massage.masseuse
    });
  }

  changeMasseuse = (event) => {
    this.setState({masseuse: event.target.value});
  }

  changeDate = (date) => {
    this.setState({date: date});
  }

  /**
   * Handles the post request.
   */
  addMassage = () => {
    Util.post(Util.MASSAGES_URL, {
      date: this.state.date.toDate(),
      masseuse: this.state.masseuse,
      client: null,
      facility: {id: this.props.facilityId}
    }, () => {
      this.props.onToggle();
      this.props.getCallback();
    });
  }

  /**
   * Handles the put request.
   */
  editMassage = () => {
    Util.put(Util.MASSAGES_URL + this.props.massage.id, {
      date: this.state.date.toDate(),
      masseuse: this.state.masseuse,
      client: this.props.massage.client,
      facility: this.props.massage.facility
    }, () => {
      this.props.onToggle();
      this.props.getCallback();
    });
  }

  render() {
    return(
      <div className='pull-right'>
        <AddButton onAdd={this.props.onToggle} />

        {this.props.active ?
          <ModalContainer onClose={this.props.onToggle}>
            <ModalDialog onClose={this.props.onToggle} width="50%">
              <h2>
                {this.props.massage === -1 ?
                  _t.translate('New Massage') : _t.translate('Edit Massage')
                }
              </h2>
              <hr />
              <div className="form-group">
                <div className="form-group">
                  <label>{ _t.translate('Masseuse') }</label>
                  <input value={this.state.masseuse} onChange={this.changeMasseuse}
                    className="form-control" />
                </div>
              </div>
              <div className="form-group">
                <label>{ _t.translate('Massage time') }</label>
                <DatePicker
                  selected={this.state.date}
                  onChange={this.changeDate}
                  dateFormat="DD. MM. HH:mm"
                  todayButton={ _t.translate('Now') }
                  className="btn btn-default"
                />
              </div>
              {this.props.massage === -1 ?
                <ModalActions
                  primaryLabel={ _t.translate('Add') }
                  onProceed={this.addMassage}
                  onClose={this.props.onToggle}
                /> :
                <ModalActions
                  primaryLabel={ _t.translate('Edit') }
                  onProceed={this.editMassage}
                  onClose={this.props.onToggle}
                />
              }
            </ModalDialog>
          </ModalContainer> : ''
        }
      </div>
    )
  }
}

export default MassageModal
