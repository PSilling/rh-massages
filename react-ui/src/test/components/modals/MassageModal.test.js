// react imports
import React from "react";
import { shallow } from "enzyme";

// test imports
import moment from "moment";
import LabeledDatetime from "../../../components/formitems/LabeledDatetime";
import LabeledInput from "../../../components/formitems/LabeledInput";
import MassageModal from "../../../components/modals/MassageModal";
import ModalActions from "../../../components/buttons/ModalActions";
import TooltipIconButton from "../../../components/iconbuttons/TooltipIconButton";
import _t from "../../../util/Translations";

// test mocks
jest.mock("../../../util/Util");

beforeAll(() => {
  Date.now = jest.fn(() => 0);
});

afterAll(() => {
  jest.resetAllMocks();
});

test("renders inside content with correct props", () => {
  const testGetFunction = jest.fn();
  const testToggleFunction = jest.fn();
  const testMasseuses = ["test"];
  const wrapper = shallow(
    <MassageModal
      active
      massage={null}
      facilityId={1}
      masseuses={testMasseuses}
      getCallback={testGetFunction}
      onToggle={testToggleFunction}
      withPortal={false}
    />
  );
  const button = wrapper.find(TooltipIconButton);
  const actions = wrapper.find(ModalActions);
  const heading = wrapper.find("h3");
  const input = wrapper.find(LabeledInput);
  const datetimes = wrapper.find(LabeledDatetime);

  expect(testToggleFunction).not.toHaveBeenCalled();
  expect(testGetFunction).not.toHaveBeenCalled();
  expect(button.props().onClick).toBe(testToggleFunction);
  expect(actions.props().onProceed).toBe(wrapper.instance().addMassage);
  expect(actions.props().onClose).toBe(testToggleFunction);
  expect(heading.props().children).toEqual(_t.translate("New Massage"));
  expect(input.props().value).toBe("");
  expect(input.props().options).toBe(testMasseuses);
  expect(datetimes.get(0).props.value).toEqual(moment("00:30", "HH:mm"));
  expect(datetimes.get(1).props.value).toEqual(moment().add(1, "hours"));
  expect(wrapper).toMatchSnapshot();
});

test("switches to edit mode when a Massage is given", () => {
  const testGetFunction = jest.fn();
  const testToggleFunction = jest.fn();
  const testMassage = {
    id: 1,
    date: new Date(0),
    ending: new Date(1000),
    masseuse: "test",
    client: null,
    facility: { id: 1, name: "test" }
  };
  const wrapper = shallow(
    <MassageModal
      active
      massage={testMassage}
      facilityId={1}
      masseuses={[]}
      getCallback={testGetFunction}
      onToggle={testToggleFunction}
      withPortal={false}
    />
  );
  wrapper.instance().setState({
    date: testMassage.date,
    time: moment.utc(moment(testMassage.ending).diff(moment(testMassage.date))),
    masseuse: testMassage.masseuse,
    massage: testMassage
  });

  const actions = wrapper.find(ModalActions);
  const heading = wrapper.find("h3");
  const input = wrapper.find(LabeledInput);
  const datetimes = wrapper.find(LabeledDatetime);

  expect(testToggleFunction).not.toHaveBeenCalled();
  expect(testGetFunction).not.toHaveBeenCalled();
  expect(heading.props().children).toEqual(_t.translate("Edit Massage"));
  expect(actions.props().primaryLabel).toBe(_t.translate("Edit"));
  expect(actions.props().onProceed).toBe(wrapper.instance().editMassage);
  expect(input.props().value).toBe("test");
  expect(datetimes.get(0).props.value).toEqual(moment.utc(moment(testMassage.ending).diff(moment(testMassage.date))));
  expect(datetimes.get(1).props.value).not.toBe(null);
});
