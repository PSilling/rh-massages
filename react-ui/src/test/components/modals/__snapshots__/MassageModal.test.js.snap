// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders inside content with correct props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MassageModal
    active={true}
    facilityId={1}
    getCallback={[MockFunction]}
    massage={null}
    masseuses={
      Array [
        "test",
      ]
    }
    onToggle={[MockFunction]}
    withPortal={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <TooltipIconButton
          disabled={false}
          icon="plus"
          onClick={[MockFunction]}
          size="md"
          style={Object {}}
          tooltip="Vytvořit novou masáž"
        />,
        <ModalBody
          tag="div"
        >
          <Row
            tag="div"
          >
            <Col
              md="12"
              tag="div"
              widths={
                Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ]
              }
            >
              <h3>
                Nová masáž
              </h3>
              <hr />
            </Col>
          </Row>
          <Row
            tag="div"
          >
            <LabeledInput
              label="Masér/Masérka"
              maxLength="64"
              onChange={[Function]}
              onEnterPress={[Function]}
              options={
                Array [
                  "test",
                ]
              }
              size="12"
              tooltip="Jméno maséra anebo masérky, který/á vykonává tuto masáž"
              type="text"
              value=""
            />
          </Row>
          <Row
            tag="div"
          >
            <LabeledDatetime
              dateFormat={false}
              disabled={false}
              label="Délka trvání"
              onChange={[Function]}
              onEnterPress={[Function]}
              size="3"
              tooltip="Jak dlouhá by měla masáž být"
              value={"1969-12-31T23:30:00.000Z"}
            />
          </Row>
          <Row
            tag="div"
          >
            <LabeledDatetime
              disabled={false}
              isValidDate={[Function]}
              label="Čas masáže"
              onChange={[Function]}
              onEnterPress={[Function]}
              size="4"
              tooltip="Kdy by se měla masáž konat"
              value={"1970-01-01T01:00:00.000Z"}
            />
          </Row>
          <ModalActions
            disabled={false}
            onClose={[MockFunction]}
            onProceed={[Function]}
            primaryLabel="Přidat"
          />
        </ModalBody>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "disabled": false,
          "icon": "plus",
          "onClick": [MockFunction],
          "size": "md",
          "style": Object {},
          "tooltip": "Vytvořit novou masáž",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Row
              tag="div"
            >
              <Col
                md="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h3>
                  Nová masáž
                </h3>
                <hr />
              </Col>
            </Row>,
            <Row
              tag="div"
            >
              <LabeledInput
                label="Masér/Masérka"
                maxLength="64"
                onChange={[Function]}
                onEnterPress={[Function]}
                options={
                  Array [
                    "test",
                  ]
                }
                size="12"
                tooltip="Jméno maséra anebo masérky, který/á vykonává tuto masáž"
                type="text"
                value=""
              />
            </Row>,
            <Row
              tag="div"
            >
              <LabeledDatetime
                dateFormat={false}
                disabled={false}
                label="Délka trvání"
                onChange={[Function]}
                onEnterPress={[Function]}
                size="3"
                tooltip="Jak dlouhá by měla masáž být"
                value={"1969-12-31T23:30:00.000Z"}
              />
            </Row>,
            <Row
              tag="div"
            >
              <LabeledDatetime
                disabled={false}
                isValidDate={[Function]}
                label="Čas masáže"
                onChange={[Function]}
                onEnterPress={[Function]}
                size="4"
                tooltip="Kdy by se měla masáž konat"
                value={"1970-01-01T01:00:00.000Z"}
              />
            </Row>,
            <ModalActions
              disabled={false}
              onClose={[MockFunction]}
              onProceed={[Function]}
              primaryLabel="Přidat"
            />,
          ],
          "tag": "div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <Col
                md="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h3>
                  Nová masáž
                </h3>
                <hr />
              </Col>,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": Array [
                  <h3>
                    Nová masáž
                  </h3>,
                  <hr />,
                ],
                "md": "12",
                "tag": "div",
                "widths": Array [
                  "xs",
                  "sm",
                  "md",
                  "lg",
                  "xl",
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "children": "Nová masáž",
                  },
                  "ref": null,
                  "rendered": "Nová masáž",
                  "type": "h3",
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {},
                  "ref": null,
                  "rendered": null,
                  "type": "hr",
                },
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <LabeledInput
                label="Masér/Masérka"
                maxLength="64"
                onChange={[Function]}
                onEnterPress={[Function]}
                options={
                  Array [
                    "test",
                  ]
                }
                size="12"
                tooltip="Jméno maséra anebo masérky, který/á vykonává tuto masáž"
                type="text"
                value=""
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "label": "Masér/Masérka",
                "maxLength": "64",
                "onChange": [Function],
                "onEnterPress": [Function],
                "options": Array [
                  "test",
                ],
                "size": "12",
                "tooltip": "Jméno maséra anebo masérky, který/á vykonává tuto masáž",
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <LabeledDatetime
                dateFormat={false}
                disabled={false}
                label="Délka trvání"
                onChange={[Function]}
                onEnterPress={[Function]}
                size="3"
                tooltip="Jak dlouhá by měla masáž být"
                value={"1969-12-31T23:30:00.000Z"}
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "dateFormat": false,
                "disabled": false,
                "label": "Délka trvání",
                "onChange": [Function],
                "onEnterPress": [Function],
                "size": "3",
                "tooltip": "Jak dlouhá by měla masáž být",
                "value": "1969-12-31T23:30:00.000Z",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <LabeledDatetime
                disabled={false}
                isValidDate={[Function]}
                label="Čas masáže"
                onChange={[Function]}
                onEnterPress={[Function]}
                size="4"
                tooltip="Kdy by se měla masáž konat"
                value={"1970-01-01T01:00:00.000Z"}
              />,
              "tag": "div",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "disabled": false,
                "isValidDate": [Function],
                "label": "Čas masáže",
                "onChange": [Function],
                "onEnterPress": [Function],
                "size": "4",
                "tooltip": "Kdy by se měla masáž konat",
                "value": "1970-01-01T01:00:00.000Z",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": null,
              "disabled": false,
              "onClose": [MockFunction],
              "onProceed": [Function],
              "primaryLabel": "Přidat",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
    ],
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <TooltipIconButton
            disabled={false}
            icon="plus"
            onClick={[MockFunction]}
            size="md"
            style={Object {}}
            tooltip="Vytvořit novou masáž"
          />,
          <ModalBody
            tag="div"
          >
            <Row
              tag="div"
            >
              <Col
                md="12"
                tag="div"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                <h3>
                  Nová masáž
                </h3>
                <hr />
              </Col>
            </Row>
            <Row
              tag="div"
            >
              <LabeledInput
                label="Masér/Masérka"
                maxLength="64"
                onChange={[Function]}
                onEnterPress={[Function]}
                options={
                  Array [
                    "test",
                  ]
                }
                size="12"
                tooltip="Jméno maséra anebo masérky, který/á vykonává tuto masáž"
                type="text"
                value=""
              />
            </Row>
            <Row
              tag="div"
            >
              <LabeledDatetime
                dateFormat={false}
                disabled={false}
                label="Délka trvání"
                onChange={[Function]}
                onEnterPress={[Function]}
                size="3"
                tooltip="Jak dlouhá by měla masáž být"
                value={"1969-12-31T23:30:00.000Z"}
              />
            </Row>
            <Row
              tag="div"
            >
              <LabeledDatetime
                disabled={false}
                isValidDate={[Function]}
                label="Čas masáže"
                onChange={[Function]}
                onEnterPress={[Function]}
                size="4"
                tooltip="Kdy by se měla masáž konat"
                value={"1970-01-01T01:00:00.000Z"}
              />
            </Row>
            <ModalActions
              disabled={false}
              onClose={[MockFunction]}
              onProceed={[Function]}
              primaryLabel="Přidat"
            />
          </ModalBody>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "icon": "plus",
            "onClick": [MockFunction],
            "size": "md",
            "style": Object {},
            "tooltip": "Vytvořit novou masáž",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <Row
                tag="div"
              >
                <Col
                  md="12"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <h3>
                    Nová masáž
                  </h3>
                  <hr />
                </Col>
              </Row>,
              <Row
                tag="div"
              >
                <LabeledInput
                  label="Masér/Masérka"
                  maxLength="64"
                  onChange={[Function]}
                  onEnterPress={[Function]}
                  options={
                    Array [
                      "test",
                    ]
                  }
                  size="12"
                  tooltip="Jméno maséra anebo masérky, který/á vykonává tuto masáž"
                  type="text"
                  value=""
                />
              </Row>,
              <Row
                tag="div"
              >
                <LabeledDatetime
                  dateFormat={false}
                  disabled={false}
                  label="Délka trvání"
                  onChange={[Function]}
                  onEnterPress={[Function]}
                  size="3"
                  tooltip="Jak dlouhá by měla masáž být"
                  value={"1969-12-31T23:30:00.000Z"}
                />
              </Row>,
              <Row
                tag="div"
              >
                <LabeledDatetime
                  disabled={false}
                  isValidDate={[Function]}
                  label="Čas masáže"
                  onChange={[Function]}
                  onEnterPress={[Function]}
                  size="4"
                  tooltip="Kdy by se měla masáž konat"
                  value={"1970-01-01T01:00:00.000Z"}
                />
              </Row>,
              <ModalActions
                disabled={false}
                onClose={[MockFunction]}
                onProceed={[Function]}
                primaryLabel="Přidat"
              />,
            ],
            "tag": "div",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <Col
                  md="12"
                  tag="div"
                  widths={
                    Array [
                      "xs",
                      "sm",
                      "md",
                      "lg",
                      "xl",
                    ]
                  }
                >
                  <h3>
                    Nová masáž
                  </h3>
                  <hr />
                </Col>,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "children": Array [
                    <h3>
                      Nová masáž
                    </h3>,
                    <hr />,
                  ],
                  "md": "12",
                  "tag": "div",
                  "widths": Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {
                      "children": "Nová masáž",
                    },
                    "ref": null,
                    "rendered": "Nová masáž",
                    "type": "h3",
                  },
                  Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "host",
                    "props": Object {},
                    "ref": null,
                    "rendered": null,
                    "type": "hr",
                  },
                ],
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <LabeledInput
                  label="Masér/Masérka"
                  maxLength="64"
                  onChange={[Function]}
                  onEnterPress={[Function]}
                  options={
                    Array [
                      "test",
                    ]
                  }
                  size="12"
                  tooltip="Jméno maséra anebo masérky, který/á vykonává tuto masáž"
                  type="text"
                  value=""
                />,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "label": "Masér/Masérka",
                  "maxLength": "64",
                  "onChange": [Function],
                  "onEnterPress": [Function],
                  "options": Array [
                    "test",
                  ],
                  "size": "12",
                  "tooltip": "Jméno maséra anebo masérky, který/á vykonává tuto masáž",
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <LabeledDatetime
                  dateFormat={false}
                  disabled={false}
                  label="Délka trvání"
                  onChange={[Function]}
                  onEnterPress={[Function]}
                  size="3"
                  tooltip="Jak dlouhá by měla masáž být"
                  value={"1969-12-31T23:30:00.000Z"}
                />,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "dateFormat": false,
                  "disabled": false,
                  "label": "Délka trvání",
                  "onChange": [Function],
                  "onEnterPress": [Function],
                  "size": "3",
                  "tooltip": "Jak dlouhá by měla masáž být",
                  "value": "1969-12-31T23:30:00.000Z",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": <LabeledDatetime
                  disabled={false}
                  isValidDate={[Function]}
                  label="Čas masáže"
                  onChange={[Function]}
                  onEnterPress={[Function]}
                  size="4"
                  tooltip="Kdy by se měla masáž konat"
                  value={"1970-01-01T01:00:00.000Z"}
                />,
                "tag": "div",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "disabled": false,
                  "isValidDate": [Function],
                  "label": "Čas masáže",
                  "onChange": [Function],
                  "onEnterPress": [Function],
                  "size": "4",
                  "tooltip": "Kdy by se měla masáž konat",
                  "value": "1970-01-01T01:00:00.000Z",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "children": null,
                "disabled": false,
                "onClose": [MockFunction],
                "onProceed": [Function],
                "primaryLabel": "Přidat",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
      ],
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
